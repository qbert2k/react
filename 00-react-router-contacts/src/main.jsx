import React from "react";
import ReactDOM from "react-dom/client";
import {createBrowserRouter, RouterProvider} from "react-router-dom";
import "./index.css";
import ErrorPage from "./error-page.jsx";
import Root, {
    loader as rootLoader,
    action as rootAction,
} from "./routes/root.jsx";
import Contact, {
    loader as contactLoader,
} from "./routes/contact.jsx";

const router = createBrowserRouter([
    {
        path: "/",
        element: <Root/>,
        errorElement: <ErrorPage/>,
        loader: rootLoader,
        action: rootAction,
        children: [
            {
                path: "contacts/:contactId",
                element: <Contact/>,
                loader: contactLoader,
            }
        ]
    }
]);

ReactDOM.createRoot(document.getElementById("root")).render(
    <React.StrictMode>
        <RouterProvider router={router}/>
    </React.StrictMode>
);