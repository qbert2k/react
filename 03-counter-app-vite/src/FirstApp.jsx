const newMessage = {
    message: 'Hola Mundo',
    title: 'title'
};

const getResult = (a, b) => a + b;

export const FirstApp = () => {
    return (<>
        <h1>{newMessage.title}</h1>
        <code>{JSON.stringify(newMessage)}</code>
        <p>Soy un texto</p>

        <p>{getResult(5, 6)}</p>
    </>);
}