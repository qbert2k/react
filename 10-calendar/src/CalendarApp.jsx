export const CalendarApp = () => {
  return (
      <div>Calendar App</div>
  );
}