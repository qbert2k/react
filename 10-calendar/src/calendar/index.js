export * from './components/CalendarEvent';
export * from './components/CalendarModal';
export * from './components/FabAddNew';
export * from './components/FabDeleteNew';
export * from './components/Navbar';
export * from './pages/CalendarPage';