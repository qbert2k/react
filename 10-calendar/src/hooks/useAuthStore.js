import {useDispatch, useSelector} from 'react-redux';
import calendarApi from '../api/calendarApi.js';
import {onChecking, onClearErrorMessage, onLogin, onLogout} from '../store/index.js';

export const useAuthStore = () => {
    const {status, user, errorMessage} = useSelector(state => state.auth);
    const dispatch = useDispatch();

    const startLogin = async ({email, password}) => {
        dispatch(onChecking());

        try {
            const {data} = await calendarApi.post('/auth', {email, password});
            localStorage.setItem('token', data.token);
            localStorage.setItem('token-init-date', new Date().getTime());
            dispatch(onLogin({name: data.name, uid: data.uid}));
        } catch (error) {
            dispatch(onLogout('Incorrect credentials'));
            setTimeout(() => {
                dispatch(onClearErrorMessage())
            }, 10);
        }
    };

    const startRegister = async ({name, email, password}) => {
        try {
            const {data} = await calendarApi.post('/auth/new', {name, email, password});
            localStorage.setItem('token', data.token);
            localStorage.setItem('token-init-date', new Date().getTime());
            dispatch(onLogin({name: data.name, uid: data.uid}));
        } catch (error) {
            dispatch(onLogout(error.response.data?.msg || '--'));
            setTimeout(()=> {
                dispatch(onClearErrorMessage())
            }, 10);
        }
    }

    return {
        // Properties
        status,
        user,
        errorMessage,

        // Methods
        startLogin,
        startRegister
    }
};