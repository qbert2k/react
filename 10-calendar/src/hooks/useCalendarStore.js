import {useDispatch, useSelector} from 'react-redux';
import {onAddNewEvent, onSetActiveEvent, onUpdateEvent} from '../store';

export const useCalendarStore = () => {
    const dispatch = useDispatch();
    const {events, activeEvent} = useSelector(state => state.calendar);

    const setActiveEvent = (calendarEvent) => {
        dispatch(onSetActiveEvent(calendarEvent));
    };

    const startSavingEvent = async (calendarEvent) => {
        if (calendarEvent._id) { // Update
            dispatch(onUpdateEvent({
                ...calendarEvent
            }));
        } else { // Create
            dispatch(onAddNewEvent({
                ...calendarEvent,
                _id: new Date().getTime()
            }));
        }
    };

    return {
        // Properties
        events,
        activeEvent,

        // Methods
        setActiveEvent,
        startSavingEvent,
    };
}