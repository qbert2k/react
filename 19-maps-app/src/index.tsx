import React from "react";
import ReactDOM from "react-dom/client";
import {MapsApp} from "./MapsApp";
import mapboxgl from "mapbox-gl";

mapboxgl.accessToken = 'pk.eyJ1IjoicWJlcnQyayIsImEiOiJjajU2d2JpOGMwdGk4MnFyMHYyb203YmJ0In0.NFxFL1P8GCDSCmBP1oZQlA';

if (!navigator.geolocation) {
    alert('The browser has not Geolocation features');
    throw new Error('The browser has not Geolocation features');
}

const root = ReactDOM.createRoot(
    document.getElementById('root') as HTMLElement
);

root.render(
    <React.StrictMode>
        <MapsApp/>
    </React.StrictMode>
);