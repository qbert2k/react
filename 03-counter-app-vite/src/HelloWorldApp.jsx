import {FirstApp} from "./FirstApp.jsx";

export const HelloWorldApp = () => {
    return (
        <h1>Hola Mundo!!!</h1>
    );
}

// function App() {
//     return <h1>Hola Mundo!!!</h1>;
// }
//
// export default App;