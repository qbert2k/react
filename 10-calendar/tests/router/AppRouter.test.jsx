import {MemoryRouter} from 'react-router-dom';
import {render, screen} from '@testing-library/react';
import {useAuthStore} from '../../src/hooks/useAuthStore';
import {AppRouter} from '../../src/router';

jest.mock('../../src/hooks/useAuthStore');

describe('Test AppRouter', () => {

    const mockCheckAuthToken = jest.fn();

    beforeEach(() => jest.clearAllMocks());

    test('should call checkAuthToken and show the loading message', () => {
        useAuthStore.mockReturnValue({
            status: 'checking',
            checkAuthToken: mockCheckAuthToken
        });

        render(<AppRouter/>);

        expect(screen.getByText('Loading...')).toBeTruthy();
        expect(mockCheckAuthToken).toHaveBeenCalled();

    });

    test('should show the login when not authenticated', () => {
        useAuthStore.mockReturnValue({
            status: 'not-authenticated',
            checkAuthToken: mockCheckAuthToken
        });

        const {container} = render(
            <MemoryRouter initialEntries={['/something/another/path']}>
                <AppRouter/>
            </MemoryRouter>
        );

        expect(screen.getByText('Log In')).toBeTruthy();
        expect(container).toMatchSnapshot();
    });
});