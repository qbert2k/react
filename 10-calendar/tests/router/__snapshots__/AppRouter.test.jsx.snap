// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test AppRouter should show the login when not authenticated 1`] = `
<div>
  <div
    class="container login-container"
  >
    <div
      class="row"
    >
      <div
        class="col-md-6 login-form-1"
      >
        <h3>
          Log In
        </h3>
        <form>
          <div
            class="form-group mb-2"
          >
            <input
              class="form-control"
              name="loginEmail"
              placeholder="Email"
              type="text"
              value=""
            />
          </div>
          <div
            class="form-group mb-2"
          >
            <input
              class="form-control"
              name="loginPassword"
              placeholder="Password"
              type="password"
              value=""
            />
          </div>
          <div
            class="d-grid gap-2"
          >
            <input
              class="btnSubmit"
              type="submit"
              value="Login"
            />
          </div>
        </form>
      </div>
      <div
        class="col-md-6 login-form-2"
      >
        <h3>
          Sign In
        </h3>
        <form>
          <div
            class="form-group mb-2"
          >
            <input
              class="form-control"
              name="registerName"
              placeholder="Name"
              type="text"
              value=""
            />
          </div>
          <div
            class="form-group mb-2"
          >
            <input
              class="form-control"
              name="registerEmail"
              placeholder="Email"
              type="email"
              value=""
            />
          </div>
          <div
            class="form-group mb-2"
          >
            <input
              class="form-control"
              name="registerPassword"
              placeholder="Password"
              type="password"
              value=""
            />
          </div>
          <div
            class="form-group mb-2"
          >
            <input
              class="form-control"
              name="registerPassword2"
              placeholder="Repeat Password"
              type="password"
              value=""
            />
          </div>
          <div
            class="d-grid gap-2"
          >
            <input
              class="btnSubmit"
              type="submit"
              value="Register"
            />
          </div>
        </form>
      </div>
    </div>
  </div>
</div>
`;
